### PR TITLE
Fid app bundle in Gradle 3.5

### LINKED ISSUE #38934: Android App Bundle does not build with Gradle 3.5.0
I tried upgrading to Gradle 3.5.0 using Android Studio, but it broke my build:

```
> flutter build appbundle
Initializing gradle...                                              2.0s
Resolving dependencies...                                          21.8s
registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
Running Gradle task 'bundleRelease'...
Note: /Users/maxlapides/flutter/.pub-cache/hosted/pub.dartlang.org/connectivity-0.4.3+7/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityPlugin.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /Users/maxlapides/flutter/.pub-cache/hosted/pub.dartlang.org/url_launcher-5.1.2/android/src/main/java/io/flutter/plugins/urllauncher/WebViewActivity.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Running Gradle task 'bundleRelease'...
Running Gradle task 'bundleRelease'... Done                       205.1s (!)
Gradle build failed to produce an Android bundle package.
```

```
❯ flutter doctor -v
[✓] Flutter (Channel beta, v1.8.3, on Mac OS X 10.14.6 18G87, locale en-US)
    • Flutter version 1.8.3 at /Users/maxlapides/flutter
    • Framework revision e4ebcdf6f4 (3 weeks ago), 2019-07-27 11:48:24 -0700
    • Engine revision 38ac5f30a7
    • Dart version 2.5.0 (build 2.5.0-dev.1.0 0ca1582afd)


[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/maxlapides/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/maxlapides/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.3, Build version 10G8
    • CocoaPods version 1.7.5

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 38.2.3
    • Dart plugin version 191.8405
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.37.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.3.0

[✓] Connected device (1 available)
    • iPhone X • 3EE2DDFA-1E6B-4E58-A867-BB24FA33DEB9 • ios • com.apple.CoreSimulator.SimRuntime.iOS-12-4 (simulator)

• No issues found!
```

### PR DESCRIPTION
**Pull request by blasten**
## Description

The Android Gradle plugin 3.5.0 generates the following app bundle: `app(-<flavor>)?-<build-type>.aab`.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/38934

## Tests

I added the following tests:

Unit tests in `gradle_test.dart`

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/android/gradle_test.dart [hunk 1]
```diff
@@ -261,6 +252,96 @@ void main() {
     }, overrides: <Type, Generator>{
       FileSystem: () => MemoryFileSystem(),
     });
+
+    testUsingContext('Finds app bundle in release mode - Gradle 3.5', () {
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/android/gradle_test.dart]: You can write helper functions to invoke a series of slightly different tests:
```dart
void genGradle35AppBundleTest(String mode) {
  testUsingContext(...);
}
genGradle35AppBundleTest('debug');
genGradle35AppBundleTest('profile');
genGradle35AppBundleTest('release');
```

Comment (blasten) [reply packages/flutter_tools/test/general.shard/android/gradle_test.dart]: I reduced the boilerplate quite a bit.

**Comment by blasten** [comment]
https://codecov.io/gh/flutter/flutter/pull/39126/diff?src=pr&el=tree#diff-cGFja2FnZXMvZmx1dHRlcl90b29scy9saWIvc3JjL2FuZHJvaWQvZ3JhZGxlLmRhcnQ= doesn't seem right

**Review by zanderso — APPROVED** [review]

**Comment by elmariocarlos** [comment]
So does this mean that we just need the master to be pushed to Stable to start using 3.5.0?

**Comment by bangonkali** [comment]
When can we get this update?

**Comment by workerbee22** [comment]
So if using Cloud based CI/CD tools, if we can't change to the master branch of Flutter we have to wait?

**Comment by elmariocarlos** [comment]
> So if using Cloud based CI/CD tools, if we can't change to the master branch of Flutter we have to wait?

Not really, if youjust put the option you want to use in the Gradle, you should be ok.

However, this should be supported soon.

**Comment by bangonkali** [comment]
Hi @elmariocarlos ,

I'm not exactly an expert in Gradle but is there a quick switch param on `flutter build` command to set this up? 

> Not really, if youjust put the option you want to use in the Gradle, you should be ok.

**Comment by elmariocarlos** [comment]
> Hi @elmariocarlos ,
> 
> I'm not exactly an expert in Gradle but is there a quick switch param on `flutter build` command to set this up?
> 
> > Not really, if youjust put the option you want to use in the Gradle, you should be ok.

Go to your android/build.gradle file, and change the version there, for example:

    dependencies {
        classpath 'com.android.tools.build:gradle:3.4.1'

**Pull request merged**